src/handler.ts

```diff
@@ -87,36 +93,67 @@ export class DebuggerHandler {
     widget: DebuggerHandler.SessionWidget[DebuggerHandler.SessionType],
     connection: Session.ISessionConnection
   ): Promise<void> {
-    const updateHandler = async () => {
+    if (!connection) {
+      delete this._kernelChangedHandlers[widget.id];
+      delete this._statusChangedHandlers[widget.id];
       return this._update(widget, connection);
+    }
+
+    const kernelChanged = () => {
+      void this._update(widget, connection);
```

Comment:
Small thing here: any reason for not returning directly?

```suggestion
      return this._update(widget, connection);
```